pkg/reloader/reloader.go

```diff
@@ -46,7 +48,7 @@ type Reloader struct {
 // If cfgEnvsubstFile is not empty, environment variables in the config file will be
 // substituted and the out put written into the given path. Prometheus should then
 // use cfgEnvsubstFile as its config file path.
-func New(logger log.Logger, reloadURL *url.URL, cfgFile string, cfgEnvsubstFile string, ruleDirs []string) *Reloader {
+func New(logger log.Logger, reloadURL *url.URL, cfgFile string, cfgEnvsubstFile string, ruleDirs []string, gunzipDir string) *Reloader {
```

Comment:
Hm.. instead of adding additional dir and making usage or reloader more complex, maybe we can either:
1. Deduce if file is compressed (and format of compression) or not
2. if yes, decompress is on the fly on reload process.

Here is how to detect if gzip was used: https://stackoverflow.com/questions/6059302/how-to-check-if-a-file-is-gzip-compressed (just check 2 first bytes).

Proposed solution is quite hard to debug, read and configure, so not changing those file paths and detecting & decompressing on the fly would be much easier for users of reload package. What do you think @wleese ?




I am pretty sure we can 1. just fine - if we cannot find

Reply:
The problem I had with that approach is that in my case we're mounting a k8s secret into a pod and using that as the configuration file. As such if it is compressed, we'd want to decompress it somewhere. We cannot write a file to the mounted secret. Also by making the output directory explicit, we can convey to the user which compression is supported through the parameter name.

Reply:
ah... that makes sense, thanks for clarifying.

Some ideas then:
* Can we change `cfgEnvsubstFile` to just `cfgOutputFile` then? And then implicitly decompress it before env substitution? That would work for you even if you don't do env var substitution.

> Also by making the output directory explicit, we can convey to the user which compression is supported through the parameter name.

This I don't get.. I guess single `cfgOutputFile` will work in this case as well, as you can set the name as you want?